internal/handlers/pg/msg_delete.go

```diff
@@ -145,17 +155,46 @@ func (h *Handler) MsgDelete(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 			return nil
 		})
 
+		return err
+	}
+
+	delErrors := new(common.WriteErrors)
+
+	var reply wire.OpMsg
+
+	// process every delete filter
+	for i := 0; i < deletes.Len(); i++ {
+		err := processQuery(i)
 		if err != nil {
-			return nil, err
+			delErrors.Append(err)
+
+			// Delete statements in the `deletes` field are not transactional.
+			// It means that we run each delete statement separately.
+			// If `ordered` is set as `true`, we don't execute the remaining statements
+			// after the first failure.
+			// If `ordered` is set as `false`,  we execute all the statements and return
+			// the list of errors corresponding to the failed statements.
+			if ordered {
+				break
+			}
 		}
 	}
 
-	var reply wire.OpMsg
+	// minimal reply document
+	replyDoc := must.NotFail(types.NewDocument(
+		"n", deleted,
+		"ok", float64(1),
```

Comment:
Is it on purpose that we always respond with `ok: 1` - even if none of docs were deleted / all the delete queries failed?

Reply:
I was also thinking about that. Tests are passing, but I will add another testcase for that.